logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/selfdeletingMessages/ObserveSelfDeletionTimerForConversationUseCase.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.selfdeletingMessages
+
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.combine
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.kaliumLogger
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlin.time.Duration.Companion.ZERO
+
+/**
+ * When invoked, this use case will start observing on a given conversation, the currently applied [SelfDeletionTimer]
+ */
+interface ObserveSelfDeletionTimerForConversationUseCase {
```

Comment:
also nit picking I suppose but the name of rather confusing to me I would expect this use case to return me some kind of self deletion timer, but all it is doing is returning settings for the self deletion timer

Reply:
Would `ObserveSelfDeletionTimerSettingsForConversationUseCase` be more clear to you? 🙃 naming is ardd

Reply:
done